src/platform/packages/shared/kbn-esql-ast/src/commands_registry/commands/rerank/autocomplete.test.ts

```diff
@@ -0,0 +1,544 @@
+  ON_KEYWORD: 'ON ',
+  PIPE: '| ',
+  COMMA_SPACE: ', ',
+  ASSIGNMENT: ' = ',
+  WITH_CLAUSE: 'WITH { $0 }',
+  AND_OPERATOR: ' AND ',
+  OR_OPERATOR: ' OR ',
+  INFERENCE_ID_KEY: '"inference_id": "$0"',
+  INFERENCE_ENDPOINT: 'inference_1',
+} as const;
+
+const OPERATOR_SUGGESTIONS = {
+  COMPARISON: ['> $0', '>= $0', '< $0', '<= $0'],
+  EQUALITY: ['!= $0', '== $0'],
+  LIKE_OPERATORS: ['LIKE $0', 'NOT LIKE $0', 'RLIKE $0', 'NOT RLIKE $0'],
+  IN_OPERATORS: ['IN $0', 'NOT IN $0'],
+  NULL_OPERATORS: ['IS NULL', 'IS NOT NULL'],
+  MATCH_OPERATOR: [': $0'],
+};
+
+const SUGGESTION_GROUPS = {
+  BASIC_CONTINUATIONS: [
+    SUGGESTION_TOKENS.COMMA_SPACE,
+    SUGGESTION_TOKENS.WITH_CLAUSE,
+    SUGGESTION_TOKENS.PIPE,
+  ] as string[],
+  FIELD_CONTINUATIONS: [
+    SUGGESTION_TOKENS.COMMA_SPACE,
+    SUGGESTION_TOKENS.ASSIGNMENT,
+    SUGGESTION_TOKENS.WITH_CLAUSE,
+    SUGGESTION_TOKENS.PIPE,
+  ] as string[],
+  BOOLEAN_CONTINUATIONS: [
+    SUGGESTION_TOKENS.COMMA_SPACE,
+    SUGGESTION_TOKENS.WITH_CLAUSE,
+    SUGGESTION_TOKENS.PIPE,
+    SUGGESTION_TOKENS.AND_OPERATOR,
+    SUGGESTION_TOKENS.OR_OPERATOR,
+  ] as string[],
+};
+
+// ============================================================================
+// Query Builder and Test Helpers
+// ============================================================================
+
+interface QueryComponents {
+  query?: string;
+  targetField?: string;
+  targetAssignment?: string;
+  onClause?: string;
+  withClause?: string;
+}
+
+const buildRerankQuery = (components: QueryComponents): string => {
+  let query = 'from a | rerank';
+
+  if (components.targetField) {
+    query += ` ${components.targetField}`;
+  }
+
+  if (components.targetAssignment) {
+    query += ` = ${components.targetAssignment}`;
+  }
+
+  if (components.query) {
+    query += ` ${components.query}`;
+  }
+
+  if (components.onClause) {
+    query += ` on ${components.onClause}`;
+  }
+
+  if (components.withClause) {
+    query += ` with ${components.withClause}`;
+  }
+
+  return query;
+};
+
+type ExpectedSuggestions = string[] | { contains?: string[]; notContains?: string[] };
+
+const expectRerankSuggestions = async (
+  query: string,
+  expected: ExpectedSuggestions,
+  mockCallbacks?: ICommandCallbacks,
+  context = mockContext,
+  offset?: number
+): Promise<void> => {
+  if (Array.isArray(expected)) {
+    return expectSuggestions(
+      query,
+      expected,
+      context,
+      'rerank',
+      mockCallbacks,
+      autocomplete,
+      offset
+    );
+  }
+
+  const results = await suggest(query, context, 'rerank', mockCallbacks, autocomplete, offset);
+  const texts = results.map((r) => r.text);
+
+  if (expected.contains?.length) {
+    expect(texts).toEqual(expect.arrayContaining(expected.contains));
+  }
+
+  if (expected.notContains?.length) {
+    expect(texts).not.toEqual(expect.arrayContaining(expected.notContains));
+  }
+};
+
+const createBooleanTestOperators = () => [
```

Comment:
Why do we create them here and we dont import them from the operators files?

Reply:
```javascript 
  describe('Terminal operators', () => {
    test.each(createBooleanTestOperators())(
      '$description behavior',
      async ({ operator, value, description }) => {
```

This is actually a testing utility with operators, expected values, and test names. I can also flat them, but they're similar tests.

Reply:
I mean that instead of rewritting them here why dont you import them from the operators file and use them to test whatever you want